build.rs

```diff
@@ -1,27 +1,46 @@
 extern crate bindgen;
+use std::{env, path::PathBuf};
 
-use std::env;
-use std::path::{PathBuf, Path};
+fn ndk_home() -> PathBuf {
+    match env::var("NDK_HOME") {
+        Ok(home) => home.into(),
+        Err(_) => {
+            let android_home: PathBuf = env::var("ANDROID_HOME")
+                .expect("Failed to get ANDROID_HOME")
+                .into();
+            android_home.join("ndk/22.1.7171670")
+        }
+    }
+}
 
-fn main() {
-    let android_home = env::var("ANDROID_HOME").expect("ANDROID_HOME not set!");
-    let ndk_include_dir = Path::new(android_home.as_str()).join("ndk/21.3.6528147/sysroot/usr/include/");
-    let gl31h = ndk_include_dir.join("GLES3/gl31.h");
-    let gl31h = gl31h.to_str().unwrap();
-    let ndk_include_dir = ndk_include_dir.to_str().unwrap();
+fn android_setup() {
+    // let ndk_include_dir = ndk_home().join("sysroot/usr/include");
+    // let gl31h = ndk_include_dir.join("GLES3/gl31.h");
+    // let gl31h = gl31h.to_str().unwrap();
+    // let ndk_include_dir = ndk_include_dir.to_str().unwrap();
 
+    // println!("{:?}", ndk_include_dir);
+    println!("cargo:rustc-link-lib=log");
+    println!("cargo:rustc-link-lib=android");
+    println!("cargo:rustc-link-lib=EGL");
     println!("cargo:rustc-link-lib=GLESv3");
-    println!("cargo:rerun-if-changed={}", gl31h);
+    // println!("cargo:rerun-if-changed={}", gl31h);
 
-    let bindings = bindgen::Builder::default()
-        .clang_arg(format!("-I{}", ndk_include_dir))
-        .header(gl31h)
-        .parse_callbacks(Box::new(bindgen::CargoCallbacks))
-        .generate()
-        .expect("Unable to generate bindings");
+    // let bindings = bindgen::Builder::default()
+    //     .clang_arg(format!("-I{}", ndk_include_dir))
+    //     .header(gl31h)
+    //     .parse_callbacks(Box::new(bindgen::CargoCallbacks))
+    //     .generate()
+    //     .expect("Unable to generate bindings");
 
-    let out_path = PathBuf::from("src");
-    bindings
-        .write_to_file(out_path.join("bindings.rs"))
-        .expect("Couldn't write bindings!");
+    // let out_path = PathBuf::from("src");
+    // bindings
+    //     .write_to_file(out_path.join("bindings.rs"))
+    //     .expect("Couldn't write bindings!");
+}
+
+fn main() {
+    if matches!(env::var("ANDROID_BUILD"), Ok(_)) {
```

Comment:
Instead of this non-standard environment variable please use something like `std::env::var("CARGO_CFG_TARGET_OS") == Ok("android".into()`

Reply:
Fixed